packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/LogObfuscator.kt

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.mongodb.internal
+
+import io.realm.kotlin.mongodb.HttpLogObfuscator
+import io.realm.kotlin.mongodb.internal.LogReplacer.Companion.defaultFeatureToReplacerMap
+
+// Replaces any given regex pattern present in a given logger message matching a number of
+// operations/features: register user, login with email and password, login with tokens and run
+// custom functions with parameters
+internal interface LogReplacer {
+
+    // Uses a replacer to obfuscate (or not) a given logger message
+    fun findAndReplace(input: String): String
+
+    companion object {
+
+        // Defines the first part of the URL path to a feature
+        private const val AUTH = "auth"
+        internal const val FUNCTIONS = "functions/call"
+
+        // Credentials feature
+        private const val CREDENTIALS_PROVIDER = "$AUTH/providers"
+
+        // Credentials type
+        private const val PROVIDER_EMAIL_PASSWORD = "$CREDENTIALS_PROVIDER/local-userpass"
```

Comment:
Nit-pick (or maybe just a way off comment 🤪), but there are awfully many constants and most of them are just used once. I think I would have strived to somehow build these from some inline `setOf(...).map(...)`, but as said, might be that it would just cause more confusion. At least this is very clean ... though a bit overkill.